### PR TITLE
[go_router_builder] Fixed the return value of the generated push method

### PR DESCRIPTION
**Pull request by vlazdra**
Starting from version 6.5.0 [[reference](https://github.com/flutter/packages/blob/main/packages/go_router/CHANGELOG.md#650)] of the **go_router** package, the method **push** is able to return a Future value.
The generator was not updated to take that into account.

This PR fixes that.

The PR resolves the reported issue: https://github.com/flutter/flutter/issues/124487

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by chunhtai — COMMENTED** [review]
Can you add a test and also bump the version?

**Comment by vlazdra** [comment]
@chunhtai Hi and thanks for the heads up! 🙌 

Regarding the package changes, I've updated the tests as well as the example in the README.
And regarding the example code that showcases the package functionality I've added a new screen that adds the return value functionality as well as added new tests for the example project.

Please let me know what you think of the changes and if I need to update anything else. 👍

**Comment by chunhtai** [comment]
@hangyujin Can you take a look at this pr?

File: packages/go_router_builder/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.2.2
+
+* Generated routes now adhere to the updated `push` method by having the ability to return the value. [go_router CHANGELOG](https://github.com/flutter/packages/blob/main/packages/go_router/CHANGELOG.md#650)
```
Comment (hangyujin) [inline packages/go_router_builder/CHANGELOG.md:3]: I think it's more consistent with other change log if we say "Supports returning value in generated push method"

**Review by hangyujin — APPROVED** [review]
LGTM

File: packages/go_router_builder/CHANGELOG.md (hunk 1 above)
Comment (vlazdra) [reply packages/go_router_builder/CHANGELOG.md]: Hey @hangyujin, thanks for the feedback. I agree. 👍

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**